File: 23boj_py/5073.py
# 삼각형과 세 변
# https://www.acmicpc.net/problem/5073

import sys

while True:
    a, b, c = map(int, sys.stdin.readline().split())
    if a == 0 and b == 0 and c == 0:
        break
    max_val = max(a, b, c)
    sum_other = a + b + c - max_val
    if max_val >= sum_other:
        print("Invalid")
    else:
        if a == b and b == c:
            print("Equilateral")
        elif a == b or b == c or a == c:
            print("Isosceles")
        else:
            print("Scalene")

File: 25boj_cpp/5073.cpp
// 삼각형과 세 변
// https://www.acmicpc.net/problem/5073

#include <bits/stdc++.h>

using namespace std;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);
  cout.tie(nullptr);

  int a, b, c;

  while (true) {
    cin >> a >> b >> c;
    if (a == 0 && b == 0 && c == 0) {
      break;
    }

    int sum = a + b + c;
    int max_val = max(a, max(b, c));
    if (sum - max_val <= max_val) {
      cout << "Invalid\n";
    } else {
      if (a == b && b == c) {
        cout << "Equilateral\n";
      } else if (a == b || b == c || a == c) {
        cout << "Isosceles\n";
      } else {
        cout << "Scalene\n";
      }
    }
  }

  return 0;
}